lib/format_parser.rb

```diff
@@ -68,9 +68,22 @@ def self.register_parser(callable_parser, formats:, natures:, priority: LEAST_PR
       end
       @parser_priorities ||= {}
       @parser_priorities[callable_parser] = priority
+
+      @registered_natures ||= []
```

Comment:
The `||=` doesn't really do what you'd expect here (not saying it doesn't work or that it isn't used above 😅 ). As a good practice, I'd see this below, when defining the reader, and then using the reader throughout the logic.

```
def self.registered_natures
    @registered_natures ||= []
end

# ...

registered_natures |= parser_provided_natures
```

(same for the other similar ones)

It also makes sense since you're anyway defining the reader for later access.

Reply:
Refactored it.